annotation/tests/test_annotations.py

```diff
@@ -83,3 +160,152 @@ def test_row_to_dict(row, expected_dictionary):
+    result = convert_bucket_name_if_s3prefix(bucket_name=bucket_name)
+    assert result == expected_string
+
+
+@pytest.mark.unittest
+@pytest.mark.parametrize(
+    [
+        "s3_provider",
+        "bucket_name",
+    ],
+    [
+        (
+            "minio",
+            TEST_TENANT,
+        ),  # successful connection
+        (
+            "aws_iam",
+            TEST_TENANT,
+        ),  # successful connection
+        (
+            "no_provider",
+            TEST_TENANT,
+        ),  # NotConfiguredException
+        (
+            "aws_iam",
+            "no_bucket",
+        ),  # NoSuchBucket
+    ],
+)
+def test_connect_s3(moto_s3, monkeypatch, s3_provider, bucket_name):
+    mock_resource = MagicMock(return_value=moto_s3)
+    monkeypatch.setattr("boto3.resource", mock_resource)
+
+    monkeypatch.setattr("annotation.annotations.main.S3_PROVIDER", s3_provider)
+
+    if s3_provider == "no_provider":
+        with pytest.raises(NotConfiguredException):
+            connect_s3(bucket_name)
+    elif bucket_name == "no_bucket":
+        with pytest.raises(moto_s3.meta.client.exceptions.NoSuchBucket):
+            connect_s3(bucket_name)
+    else:
+        result_s3 = connect_s3(bucket_name)
+        assert moto_s3 == result_s3
+
+
+@pytest.mark.unittest
+def test_get_sha_of_bytes():
+    assert sha1(b"1").hexdigest() == get_sha_of_bytes(b"1")
+
+
+@pytest.mark.unittest
+@pytest.mark.parametrize(
+    [
+        "annotated_doc",
+        "s3_path",
+        "phys_path",
+        "bucket_physical",
+        "expected_manifest",
+    ],
+    [
+        (
+            ANNOTATION_DOC_CATEGORIES,
+            f"annotation/{str(ANNOTATION_DOC_CATEGORIES.job_id)}/"
+            f"{str(ANNOTATION_DOC_CATEGORIES.file_id)}",
+            "path/to/file",
+            "bucket-of-phys-file",
+            {
+                "pages": {},
+                "validated": [1],
+                "failed_validation_pages": [],
+                "file": "path/to/file",
+                "bucket": "bucket-of-phys-file",
+                "categories": [
+                    {"type": "taxonomy", "value": "foo"},
+                    {"type": "taxonomy", "value": "bar"},
+                ],
+            },
+        )
+    ],
+)
+def test_create_manifest_json(
+    moto_s3,
+    mock_db_query_filter_order_by_all_empty,
+    annotated_doc,
+    s3_path,
+    phys_path,
+    bucket_physical,
+    expected_manifest,
+):
+
+    with patch(
+        "annotation.annotations.main.accumulate_pages_info",
+        return_value=[{1: "validated"}, {}, {}],
+    ) as mock_accumulate_pages_info, patch(
+        "annotation.annotations.main.row_to_dict"
+    ) as mock_row_to_dict, patch(
+        "annotation.annotations.main.upload_json_to_minio"
+    ) as mock_upload_json_to_minio:
+        create_manifest_json(
+            annotated_doc,
+            s3_path,
+            phys_path,
+            bucket_physical,
+            ANNOTATION_DOC_CATEGORIES.tenant,
```

Comment:
You can use annotated_doc instead of the global object ANNOTATION_DOC_CATEGORIES, as that is the purpose of parameterized tests. The purpose of parameterized tests is to utilize the passed parameters within the function.

Reply:
I used `ANNOTATION_DOC_CATEGORIES `for easier traversal and readability of the test case, since` Ctrl+Click` on an `annotated_doc `doesn't send you straight to the parameter.